### PR TITLE
BarGauge: First bar gauge panel option & some refactoring

### PR DESCRIPTION
**Pull request by torkelo**
First migration of panel options to the new options builder. 

* Fixed bug with "Auto" orientation that required a change to VizRepeater so that it passes the orientation (it handled the auto logic and do not want to duplicate it) 
* Renamed the functions on PanelPlugin (feedback wanted)

### CODE REVIEW

File: packages/grafana-data/src/types/panel.ts [hunk 1]
```diff
@@ -120,8 +120,9 @@ export interface PanelOptionsEditorItem<TValue = any, TSettings = any> extends R
   settings?: TSettings;
 }
 
-export interface PanelOptionsEditorConfig<TSettings = any, TValue = any>
-  extends Pick<PanelOptionsEditorItem<TValue, TSettings>, 'id' | 'description' | 'name' | 'settings'> {}
+export interface PanelOptionsEditorConfig<TSettings = any> extends RegistryItem {
```
Comment (torkelo) [inline packages/grafana-data/src/types/panel.ts]: @dprokop felt this was simpler and seems to work the Pick that only picks the props from RegistryItem felt a bit fragile

Comment (dprokop) [reply packages/grafana-data/src/types/panel.ts]: Fragile in what way? I am fine with this change tho, the only thing i don't like is these two properties will be available when using the API: 
```
  aliasIds?: string[]; // when the ID changes, we may want backwards compatibility ('current' => 'last')
  /**
   * Some extensions should not be user selectable
   *  like: 'all' and 'any' matchers;
   */
  excludeFromPicker?: boolean;```

Not that it's a big deal, but i like to expose only the things that really matter.

**Review by dprokop — APPROVED** [review]
I think the renames are fine. Been struggling a lot with finding good names for those methods, I think the ones proposed by you are better. Btw, I've updated docs comments as those were not updated. 

Other than that - LGTM!

File: packages/grafana-data/src/types/panel.ts (hunk 1 above)
Comment (torkelo) [reply packages/grafana-data/src/types/panel.ts]: then I think we should just add those properties to this type, and not extend RegistyItem

**Pull request merged**